### DISCUSSION TITLE
Audio/Video Codec/Channels column in episode list

### DISCUSSION THREAD

**Discussion by bnevets27** [pymedusa/Medusa#5114] [Ideas & Feature requests]
**Is your feature request related to a problem? Please describe.**
I would like to be able to differentiate files that have 2.0 audio and files that have 5.1 audio.

**Describe the solution you'd like**
A column that shows audio codec/channels.

**Describe alternatives you've considered**
Puting channel info into the file name. This requires the info be present in the release name with it isn't always there nor always accurate. And I personally don't want that info in my file name.

**Additional context**
Personally would be really helpful to see the audio channels a file has. Yes it is included in the release name and could be added during renaming but if the file doesn't say then its not possible. Releases that don't say they have 5.1 can still have 5.1 audio. 

Full disclaimer yes I saw this in sonarr v3

**Comment by p0psicles** [comment]
@pymedusa/developers we could provide an intermediate solution to this, by exposing the knowit information, stored in dogpile?
Thoughts?

**Comment by OmgImAlexis** [comment]
@p0psicles `/file/` endpoint? We could expose all the data about each file via knowit + guessit so third-party apps can do whatever they want with it.

**Comment by p0psicles** [comment]
@OmgImAlexis I just checked. But knowit is already part of apiv2 spec. through `/series/{seriesid}/episodes/{id}/metadata`

Only thing we should think about now, is to dogpile it and start using it somehow.

**Comment by p0psicles** [comment]
Created a small POC, where i'm dogpiling knowit information for an episode.
Btw, the apiv2 works fine.

But, every time your call the api on an episode, it scans the file. So I don't know if you want to keep doing that. That's why I want to cache it first, before starting to add it to the UI.

**Comment by OmgImAlexis** [comment]
@p0psicles maybe we should add a `refresh=true` param where it'd grab it from your new cache if `false` otherwise it'd grab it from the file itself.

We could probably use this in a way to replace the "refresh" button in `displayShow`. It could return the new data as well as if the file exists or has been removed, etc. meaning we could update the season table.

Thoughts?

**Comment by p0psicles** [comment]
My proposal is isolated to knowit metadata. So all the metadata thats not stored in the tv_episodes table. So thats semi-structured data.

Sure the release name parsing results can also be stored in dogpile. But im not sure if we should switch to dogpile for this. 

At that point we should evaluate a nosql db or at least sqlalchemy.
